docs/development/when_to_rebase.rst

```diff
@@ -0,0 +1,45 @@
+*********************************
+When to rebase and squash commits
+*********************************
+
+This page describes recommendations for when to rebase pull requests and when to
+combine/squash commits.
+
+When to remove or combine/squash commits
+========================================
+
+Pull requests **must** be rebased and at least partially squashed (but not
+necessarily squashed to a single commit) if large (approximately >10KB)
+non-source code files (e.g. images, data files, etc.) are added and then removed
+or modified in the PR commit history (The squashing should remove all but the
+last addition of the file to not use extra space in the repository).
+
+Combining/squashing commits is **encouraged** but not required where the number
+of commits is excessive for the changes made.  The definition of 'excessive' is
+subjective, but would never apply for 5 or fewer commits even for a trivial
+change.  Reviewers should use their discretion in suggesting to combine/squash
+commits.  Be mindful of maintaining a welcoming environment and be helpful with
+advice, especially for new contributors.
+
+As an example, for a change affecting < 10 lines of source code and including a
```

Comment:
Feature request: Add a function to calculate this in `astropy-tools`. For example: `pr_has_excessive_commits(...)` would return `True` or `False`. Maintainer can enter those number manually for calculations, so GitHub API integration is not needed. p.s. Given the subjective nature of such calculation, you can say no...

Reply:
hmm, this is very subjective, and I wouldn't push a newcomer to squash in borderline cases (e.g. ~one commit changes 5 lines, the other one adds a changelog is fine in my reading, having 5 doing the 5 lines is not).